doc/testing.adoc

```diff
@@ -0,0 +1,60 @@
+== Testing without external broker
+
+It's not rare to have to test your application but deploying the infrastructure can be cumbersome.
+While Docker or Test Containers have improved the the testing experience, you may want to mock this infrastructure.
+
+SmallRye Reactive Messaging proposes an _in-memory_ connector for this exact purpose.
+It allows switching the connector used for a channel with an _in-memory_ connector.
+This connector can then provide a way to send messages for incoming channels, or checked the received messages for outgoing
+channels.
+
+For example, in a test, you can do something like:
+
+[source, java]
+----
+// ...
+import io.smallrye.reactive.messaging.connector.InMemoryConnector;
+import io.smallrye.reactive.messaging.connector.InMemorySource;
+import io.smallrye.reactive.messaging.connector.InMemorySink;
+// ...
+
+@Test
+class MyTest {
+
+    // 1. Switch the channels to the in-memory connector:
+    @BeforeAll
+    public static void switchMyChannels() {
+        InMemoryConnector.switchIncomingChannelToInMemory("prices");
```

Comment:
Does it make sense for this to be a more generic `switchChannelToInMemory()`? Or it needs to specifically refer to incoming and outgoing?

For instance, does it make sense to set it for both?

Reply:
unfortunately no as it would configure for both incoming and outgoing which may lead to conflicts with the rest of the application.

Reply:
Would there be conflicts? If you're mocking a set of channels, you're not likely using real channels for the others are you?

So if "prices" is mocked for incoming, but not outgoing, that would prevent a method being used to create the values for the incoming channel, unless you explicitly mocked the outgoing as well?

Wondering if we can just presume everything is being mocked, and therefore incoming or outgoing is irrelevant, we're just switching a channel name to "in memory"?

Reply:
Actually, I need to try, it may work by accident.

Reply:
It does work :-) Forgot one small details about the spec: channel names are unique.
Updated.